pkg/appcommon/harness_test.go

```diff
@@ -64,3 +67,34 @@ func TestApp_Close(t *testing.T) {
 	})
 
 }
+
+func TestApp_Config_Tracer(t *testing.T) {
+	t.Run("tracer from config is set as global tracer", func(t *testing.T) {
+		defer tearDown(t)
+
+		tracer := mocktracer.New()
+		app, err := New(Config{ServiceName: "test", Tracer: tracer, InstrumentBuckets: "0.1"}, prometheus.NewRegistry(), "")
+		require.NoError(t, err)
+		require.Equal(t, tracer, opentracing.GlobalTracer())
+
+		err = app.Close()
+		require.NoError(t, err)
+	})
+
+	t.Run("new global tracer created if config tracer is nil", func(t *testing.T) {
+		defer tearDown(t)
+
+		app, err := New(Config{ServiceName: "test", Tracer: nil, InstrumentBuckets: "0.1"}, prometheus.NewRegistry(), "")
+		require.NoError(t, err)
+		require.NotNil(t, opentracing.GlobalTracer())
+
+		err = app.Close()
+		require.NoError(t, err)
+	})
+}
+
+func tearDown(t *testing.T) {
```

Comment:
since this teardown only applies to the single test, can you name it something like tearDownConfigTracerTest or make it inline in the test?

Reply:
👍